dali/pipeline/data/tensor_list_as_tensor_vector_test.cc

```diff
@@ -0,0 +1,182 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+
+#include "dali/core/access_order.h"
+#include "dali/core/common.h"
+#include "dali/core/format.h"
+#include "dali/core/tensor_layout.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/data/buffer.h"
+#include "dali/pipeline/data/tensor_vector.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/test/dali_test.h"
+#include "dali/test/tensor_test_utils.h"
+
+namespace dali {
+namespace test {
+
+/***************************************************************************************************
+ The section below is copied from tensor_list_test.cc with the only changes being:
+ * Rename all occurrences of TensorList to TensorVector
+ * Remove APIs that will be not ported: tensor_offset
+ * Tests were generalized over contiguous/noncontiguous option (see the pairs in
+   TensorVectorBackendContiguous)
+ * SetContiguous(true) was added in few places, as there is a difference in defaults between
+   current TensorVector and TensorList.
+ **************************************************************************************************/
+
+template <typename TypeParam>
+class TensorVectorTest : public DALITest {
+ public:
+  using Backend = std::tuple_element_t<0, TypeParam>;
+  static constexpr bool kState = std::tuple_element_t<1, TypeParam>::value;
+
+  TensorListShape<> GetRandShape() {
+    int num_tensor = this->RandInt(1, 64);
+    int dims = this->RandInt(2, 3);
+    TensorListShape<> shape(num_tensor, dims);
+    for (int i = 0; i < num_tensor; ++i) {
+      TensorShape<> tensor_shape;
+      tensor_shape.resize(dims);
+      for (int j = 0; j < dims; ++j) {
+        tensor_shape[j] = this->RandInt(1, 200);
+      }
+      shape.set_tensor_shape(i, tensor_shape);
+    }
+    return shape;
+  }
+
+  TensorListShape<> GetSmallRandShape() {
+    int num_tensor = this->RandInt(1, 32);
+    int dims = this->RandInt(2, 3);
+    TensorListShape<> shape(num_tensor, dims);
+    for (int i = 0; i < num_tensor; ++i) {
+      TensorShape<> tensor_shape;
+      tensor_shape.resize(dims);
+      for (int j = 0; j < dims; ++j) {
+        tensor_shape[j] = this->RandInt(1, 64);
+      }
+      shape.set_tensor_shape(i, tensor_shape);
+    }
+    return shape;
+  }
+
+  /**
+   * Initialize & check a TensorVector based on an input shape
+   * Allocate it as float
+   */
+  void SetupTensorVector(TensorVector<Backend> *tensor_list,
+                       const TensorListShape<>& shape,
+                       vector<Index> *offsets) {
```

Comment:
I think we've abandoned using pointers instead of references for non-nullable output parameters.
```suggestion
  void SetupTensorVector(TensorVector<Backend> &tensor_list,
                         vector<Index> &offsets,
                         const TensorListShape<> &shape) {
```

Reply:
As the comment states:
>  The section below is copied from tensor_list_test.cc with the only changes being:
>  * Rename all occurrences of TensorList to TensorVector
I have the full test file ported in another PR, but here I just copied the relevant tests.

Reply:
aligned